### ISSUE TITLE
CRAN: fix noLD errors

### ISSUE THREAD

**Issue by bschilder** [neurogenomics/rworkflows#124]
Got a sassy email from CRAN:
>Please correct before 2024-01-10 to safely retain your package on CRAN.
> It seems we need to remind you of the CRAN policy:
>'Packages which use Internet resources should fail gracefully with an informative message
if the resource is not available or has changed (and not give a check warning nor error).'
>This needs correction whether or not the resource recovers.
>The CRAN Team

https://www.stats.ox.ac.uk/pub/bdr/noLD/rworkflows.out

CRAN's servers have issues with internet connectivity which can cause problems with getting remote resources during checks. Will write some safeguards to avoid errors, even though I think producing an error is the correct outcome when resources can't be reached.

**Comment by bschilder** [comment]
Here's the specific failed test:
```R
══ Failed tests ════════════════════════════════════════════════════════════════
  ── Failure ('test-check_cont.R:26:3'): check_cont works ────────────────────────
  Expected `check_cont(cont = "docker.io/bioconductor/bioconductor_docker:devel")` to run without any warnings.
  i Actually got a <simpleWarning> with text:
    URL 'https://hub.docker.com/v2/repositories//bioconductor/bioconductor_docker/tags?page_size=1000': status was 'Couldn't resolve host name'
  
  [ FAIL 1 | WARN 0 | SKIP 0 | PASS 182 ]
  Error: Test failures
  Execution halted
```

**Comment by bschilder** [comment]
Found some super useful functions for resolving these issues without disabling the tests on GHA (which has relatively more reliable internet):
```R
  testthat::skip_on_cran()
  testthat::skip_if_offline()
```

**Comment by bschilder** [comment]
CRAN:
> Please correct before 2024-01-10 to safely retain your package on CRAN.

Also CRAN:
<img width="1052" alt="Screenshot 2023-12-29 at 13 37 14" src="https://github.com/neurogenomics/rworkflows/assets/34280215/35beac86-7bba-4693-94ef-bce6f3c73e5d">

I'm emailing CRAN to request an extension as I will be away on holiday from Jan 6-13.

**Comment by bschilder** [comment]
CRAN never replied to my email, and today I just found out that they went ahead and deprecated rworkflows anyway:
https://cran.r-project.org/web/packages/rworkflows/index.html
![Screenshot 2024-01-18 at 16 14 49](https://github.com/neurogenomics/rworkflows/assets/34280215/e589b099-1af1-48ea-9332-e3acbef6b03c)

Not cool, CRAN. Not cool at all....

I've emailed them again asking them to un-deprecate it given the circumstances. In tandem, I'm submitting the new version.

**Comment by NathanSkene** [comment]
That is ridiculous, great timing from them

From: Brian M. Schilder ***@***.***>
Date: Thursday, 18 January 2024 at 16:16
To: neurogenomics/rworkflows ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [neurogenomics/rworkflows] CRAN: fix noLD errors (Issue #124)
This email from ***@***.*** originates from outside Imperial. Do not click on links and attachments unless you recognise the sender. If you trust the sender, add them to your safe senders list<https://spam.ic.ac.uk/SpamConsole/Senders.aspx> to disable email stamping for this address.



CRAN never replied to my email, and today I just found out that they went ahead and deprecated rworkflows anyway:
https://cran.r-project.org/web/packages/rworkflows/index.html
Screenshot.2024-01-18.at.16.14.49.png (view on web)<https://github.com/neurogenomics/rworkflows/assets/34280215/e589b099-1af1-48ea-9332-e3acbef6b03c>

Not cool, CRAN. Not cool at all....

I've emailed them again asking them to un-deprecate it given the circumstances. In tandem, I'm submitting the new version.

**Comment by bschilder** [comment]
I asked them to de-deprecate rworkflows given the situation and they replied by refusing to. Instead, they ignored their role in the situation that led to its deprecation and asked me to submit a new version (which I had already done and was waiting for them to approve).

No reply on why I didn't get a notification that rworkflows had been removed from CRAN (or any additional warning leading up to its deprecation).

Since then, my new version of rworkflows is back up on CRAN, but now one of their server is failing due to pandoc not being properly installed on it (an issue that has come up previously). I've emailed them to alert them of the issue.
